src/analysis/liouville/inequalities_and_series.lean

```diff
@@ -0,0 +1,122 @@
+      ((one_div_lt (zero_lt_one.trans hm) zero_lt_one).mpr (one_div_one.le.trans_lt hm))),
+  rw [div_pow, one_pow],
+  refine (one_div_le_one_div _ _).mpr (pow_le_pow hm.le (fi a));
+  exact pow_pos (zero_lt_one.trans hm) _
+end
+
+end lemmas_about_summability_and_sums
+
+/--  Partial inequality, works with `m ∈ ℝ` satisfying `1 < m`. -/
+lemma tsum_one_div_pow_factorial_lt (m1 : 1 < m) (n : ℕ) :
+  ∑' (i : ℕ), 1 / m ^ (i + (n + 1))! < (1 - 1 / m)⁻¹ * (1 / m ^ (n + 1)!) :=
+-- two useful inequalities
+have m0 : 0 < m := (zero_lt_one.trans m1),
+have mi : abs (1 / m) < 1 :=
+  (le_of_eq (abs_of_pos (one_div_pos.mpr m0))).trans_lt ((div_lt_one m0).mpr m1),
+calc (∑' i, 1 / m ^ (i + (n + 1))!)
+    < ∑' i, 1 / m ^ (i + (n + 1)!) :
+    -- to show the strict inequality between these series, we prove that:
+    tsum_lt_tsum_of_nonneg
+      -- 1. the first series has non-negative terms
+      (λ b, one_div_nonneg.mpr (pow_nonneg m0.le _))
+      -- 2. the second series dominates the first
+      (λ b, one_div_mono_exp m1.le (b.add_factorial_succ_le_factorial_add_succ n))
+      -- 3. the term with index `i = 2` of the first series is strictly smaller than
+      -- the corresponding term of the second series
+      (one_div_pow_strict_mono m1 (n.add_factorial_succ_lt_factorial_add_succ rfl.le))
+      -- 4. the second series is summable, since its terms grow quickly
+      (summable_inv_pow_ge m1 (λ j, nat.le.intro rfl))
+... = ∑' i, (1 / m) ^ i * (1 / m ^ (n + 1)!) :
+    -- split the sum in the exponent and massage
+    by { congr, ext i, rw [pow_add, ← div_div_eq_div_mul, div_eq_mul_one_div, ← one_div_pow i] }
+-- factor the constant `(1 / m ^ (n + 1)!)` out of the series
+... = (∑' i, (1 / m) ^ i) * (1 / m ^ (n + 1)!) : tsum_mul_right
+... = (1 - 1 / m)⁻¹ * (1 / m ^ (n + 1)!) :
+    -- the series if the geometric series
+    mul_eq_mul_right_iff.mpr (or.inl (tsum_geometric_of_abs_lt_1 mi))
+
+lemma liouville.aux_calc (n : ℕ) (hm : 2 ≤ m) :
+  (1 - 1 / m)⁻¹ * (1 / m ^ (n + 1)!) ≤ 1 / (m ^ n!) ^ n :=
+calc (1 - 1 / m)⁻¹ * (1 / m ^ (n + 1)!) ≤ 2 * (1 / m ^ (n + 1)!) :
+  -- the second factors coincide (and are non-negative),
+  -- the first factors, satisfy the inequality `sub_one_div_inv_le_two`
+  mul_mono_nonneg (one_div_nonneg.mpr (pow_nonneg (zero_le_two.trans hm) _))
+    (sub_one_div_inv_le_two hm)
+... = 2 / m ^ (n + 1)! : mul_one_div 2 _
+... = 2 / m ^ (n! * (n + 1)) : congr_arg ((/) 2) (congr_arg (pow m) (mul_comm _ _))
+... ≤ 1 / m ^ (n! * n) :
+  begin
+    -- [ NB: in this block, I did not follow the brace convention for subgoals.  The
+    --   reason is that all extraneous goals are solved by
+    --   `exact pow_pos (zero_lt_two.trans_le hm) _` and are created also by later tactics.
+    --   Thus, I waited until the last tactic producing a repeated goal and then solve them
+    --   all at once using `any_goals { exact pow_pos (zero_lt_two.trans_le hm) _ }`. ]
```

Comment:
not sure how needed this massive explanation is :b

Reply:
Eric, thank you very much!  I am going to leave the comments, since I find them helpful, when I see them.  I did shorten the one that you highlighted, though!